lib/routes/favorite/favorite.controller.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const Course = require('../../models/course.js');
+
+// function gets all favorites of the current user
+exports.getAllFavorites = function(req, res) {
+
+  getAllFavoritesHelper(req.user)
+  .then(function(result) {
+    if (result === false) {
+      res.status(500).send('Server Error.');
+    } else {
+      res.send(result.toJSON());
+    }
+  });
+
+};
+
+// function that does the non-HTTP work of getAllFavorites
+// useful for testing
+function getAllFavoritesHelper(user) {
+  return new Promise(function(resolve, reject) {
+    user.fetch({withRelated: ['favorites']})
+    .then(function(userData) {
+      resolve(userData.related('favorites'));
+    }).catch(function(err) {
```

Comment:
Should also `.catch(User.NotFoundError, function () {})` and send a `404` status with an error message.

Same as this: https://github.com/coursereviews/schedule/blob/master/lib/routes/extracurricular/extracurricular.controller.js#L18-L20